source/tool/chipsec/hal/acpi.py

```diff
@@ -371,14 +371,55 @@ def get_ACPI_table_list( self ):
         # cache RSDT/XSDT in the list of ACPI tables
         if sdt_pa is not None: self.tableList[ sdt_header.Signature ].append(sdt_pa)
 
-        # cache other ACPI tables in the list
+        self.get_table_list_from_SDT(sdt, is_xsdt)
+
+        if ACPI_TABLE_SIG_FACP in self.tableList:
+            self.get_DSDT_from_FADT()
+        else:
+            logger().warn( 'Cannot find FADT in %s' % ('XSDT' if is_xsdt else 'RSDT') )
+
+        return self.tableList
+
+    #
+    # Gets table list from entries in RSDT/XSDT
+    #
+    def get_table_list_from_SDT(self, sdt, is_xsdt):
+        logger().log( 'Getting table list from entries in %s' % ('XSDT' if is_xsdt else 'RSDT') )
         for a in sdt.Entries:
             _sig = self.cs.mem.read_physical_mem( a, ACPI_TABLE_SIG_SIZE )
             if _sig not in ACPI_TABLES.keys():
                 logger().warn( 'Unknown ACPI table signature: %s' % _sig )
             self.tableList[ _sig ].append(a)
 
-        return self.tableList
+    #
+    # Gets DSDT from FADT
+    #
+    def get_DSDT_from_FADT(self):
+        logger().log( 'Getting DSDT from FADT' )
+
+        (_, parsed_fadt_content, _, _) = self.get_parse_ACPI_table('FACP')[0]
+        dsdt_address = parsed_fadt_content.dsdt
+        x_dsdt_address = parsed_fadt_content.x_dsdt
+        dsdt_address_to_use = None
+
+        if x_dsdt_address is None:

```

Comment:
I am wondering if this logic should be part of the FADT class. Maybe add a method to decide which SDT to use?